controller/api/public/grpc_server.go

```diff
@@ -446,6 +532,34 @@ func (s *grpcServer) query(ctx context.Context, queryReq telemPb.QueryRequest) q
 	return queryResult{res: *queryRsp, err: nil}
 }
 
+func (s *grpcServer) shouldIgnore(pod *k8sV1.Pod) bool {
+	for _, namespace := range s.ignoredNamespaces {
+		if pod.Namespace == namespace {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *grpcServer) getPodOwner(pod *k8sV1.Pod) (string, error) {
```

Comment:
I think it would better for this method to only return successfully if the owner object is actually a deployment. As written this will return replication controllers, jobs, etc. The new stat API only treats actual deployments as deployments, and since these two APIs work in conjunction together, I'd like for them to be consistent. I think something like this would work:

```
func (s *grpcServer) getDeploymentFor(pod *k8sV1.Pod) (string, error) {
	namespace := pod.Namespace
	if len(pod.GetOwnerReferences()) == 0 {
		return "", fmt.Errorf("Pod %s has no owner", pod.Name)
	}
	parent := pod.GetOwnerReferences()[0]
	if parent.Kind != "ReplicaSet" {
		return "", fmt.Errorf("Pod %s parent is not a ReplicaSet", pod.Name)
	}
	rs, err := s.replicaSetLister.GetPodReplicaSets(pod)
	if err != nil {
		return "", err
	}
	if len(rs) == 0 || len(rs[0].GetOwnerReferences()) == 0 {
		return "", fmt.Errorf("Pod %s has no replicasets", pod.Name)
	}

	for _, r := range rs {
		for _, owner := range r.GetOwnerReferences() {
			switch owner.Kind {
			case "Deployment":
				return namespace + "/" + owner.Name, nil
			}
		}
	}

	return "", fmt.Errorf("Pod %s owner is not a Deployment")
}
```

Reply:
Makes sense, thanks!